src/main/java/edu/yale/its/tp/cas/client/filter/CASFilter.java

```diff
@@ -22,6 +22,7 @@
 import javax.servlet.http.HttpServletResponse;
 import javax.servlet.http.HttpSession;
 
+import edu.yale.its.tp.cas.util.Configuration;

```

Comment:
so `isReceiptQueuedForLogout()` checks for the service ticket from the `CASReceipt` (from the session), not from the http request.  So it should work. Perhaps, though, I'm misunderstanding your question?